go/adbc/driver/flightsql/utils.go

```diff
@@ -71,5 +73,7 @@ func adbcFromFlightStatus(err error) error {
 	return adbc.Error{
 		Msg:  err.Error(),
 		Code: adbcCode,
+		// slice of proto.Message or error
+		Details: grpcStatus.Details(),
```

Comment:
it looks like the details themselves are effectively `proto.Message` interfaces that we could leverage?

Reply:
Yes, that's why I made it interface{} so we can easily return them directly without forcing apps to serialize/deserialize them

Reply:
ideally i'd prefer to be explicit on the interface type and use `proto.Message` rather than `interface{}`

Reply:
Trying to wrap this up, now it's behind an interface with wrappers for Protobuf messages.